### PR TITLE
fix(database-change): supports the execution of sql by downloading files from a separate bucket

### PR DESCRIPTION
**Pull request by kiko-art**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
sql plan task supports the execution of sql by downloading files from a separate bucket
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
 import org.springframework.jdbc.core.ConnectionCallback;
 import org.springframework.jdbc.core.StatementCallback;
 
+import com.amazonaws.util.CollectionUtils;
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java:32]: Use apache commons instead.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java [hunk 2]
```diff
@@ -163,6 +164,47 @@ protected boolean doStart(JobContext context) throws Exception {
...
     }
 
+    private List<OffsetString> getSplitSql() {
+        SqlCommentProcessor processor = ConnectionSessionUtil.getSqlCommentProcessor(connectionSession);
+        StringBuffer buffer = new StringBuffer();
+        List<OffsetString> splitSqlList = new ArrayList<>();
+
+        if (CollectionUtils.isNullOrEmpty(parameters.getSqlObjectIds())) {
+            String sqlContent = parameters.getSqlContent();
+            log.info("SQL content={}", sqlContent);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java:174]: sqlContent may too long for log output

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java [hunk 3]
```diff
@@ -163,6 +164,47 @@ protected boolean doStart(JobContext context) throws Exception {
...
+
+            List<OffsetString> offsetStringList = processor.split(buffer, sqlContent);
+            result.setTotalStatements(offsetStringList.size());
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java:177]: result.setTotalStatements may set out of getSplitSql method, responsibility should isolated

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java [hunk 4]
```diff
@@ -163,6 +164,47 @@ protected boolean doStart(JobContext context) throws Exception {
...
+            try {
+                byte[] bytes = cloudObjectStorageService.readContent(objectId);
+                if (Objects.isNull(bytes)) {
+                    log.warn("Read content from cloud storage returned null, objectId={}", objectId);
+                    continue;
+                }
+                log.info("successfully read content from cloud storage, objectId={}", objectId);
+                String sqlContent = new String(bytes, StandardCharsets.UTF_8);
+                List<OffsetString> offsetStringList = processor.split(buffer, sqlContent);
+                splitSqlList.addAll(offsetStringList);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java:199]: read all sql statement into memory may cost a lot of memory, 
the database change task already contains stream read/split logic, sql plan may follow the same flow.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java (hunk 1 above)
Comment (kiko-art) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java]: Already changed.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java (hunk 4 above)
Comment (kiko-art) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java]: Already changed,  the sql plan follow the same logic as stream reading.

**Review by yizhouxw — DISMISSED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]

**Review by guowl3 — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**